### ISSUE TITLE
Ctrl+Super causes stack smashing crash in kitty

### ISSUE THREAD

**Issue by anasem** [neovim/neovim#14894]
- `nvim --version`:

```
NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.4/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```

- Operating system/version: Linux 5.12.9
- Terminal name/version: kitty 0.20.3
- `$TERM`: xterm-kitty

[ ] `vim -u DEFAULTS` (version: ) behaves differently

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
# Alternative for shell-related problems:
# env -i TERM=ansi-256color "$(which nvim)"

```

1. Open vim
2. Press `Ctrl+Super(win key)`

### Actual behaviour
```
*** stack smashing detected ***: terminated
```

Backtrace
```
Stack trace of thread 1842632:
#0  0x00007fd05a990d22 raise (libc.so.6 + 0x3cd22)
#1  0x00007fd05a97a862 abort (libc.so.6 + 0x26862)
#2  0x00007fd05a9d2d28 __libc_message (libc.so.6 + 0x7ed28)
#3  0x00007fd05aa61cca __fortify_fail (libc.so.6 + 0x10dcca)
#4  0x00007fd05aa61c94 __stack_chk_fail (libc.so.6 + 0x10dc94)
#5  0x000055f77a0fecab input_enqueue (nvim + 0x1fdcab)
#6  0x000055f77a1a2bcb n/a (nvim + 0x2a1bcb)
#7  0x000055f77a013ca5 loop_poll_events (nvim + 0x112ca5)
#8  0x000055f77a0f7a12 n/a (nvim + 0x1f6a12)
#9  0x000055f77a0fedb9 os_inchar (nvim + 0x1fddb9)
#10 0x000055f77a17b51f state_enter (nvim + 0x27a51f)
#11 0x000055f77a0c5c2c normal_enter (nvim + 0x1c4c2c)
#12 0x000055f779f6a375 main (nvim + 0x69375)
#13 0x00007fd05a97bb25 __libc_start_main (libc.so.6 + 0x27b25)
#14 0x000055f779f6b1de _start (nvim + 0x6a1de)
```

**Comment by glacambre** [comment]
Hi, thanks for reporting this crash, can you reproduce this on the master branch/nightly release?

**Comment by anasem** [comment]
Hi, thanks for the reply! Indeed, it seems fixed on master `V0.5.0-dev`
